pkg/backup/compaction_job.go

```diff
@@ -174,8 +224,9 @@ func StartCompactionJob(
 		return 0, errors.New("missing job execution context")
 	}
 	details := jobspb.BackupDetails{
-		StartTime: start,
-		EndTime:   end,
+		ScheduleID: scheduleID,
+		StartTime:  start,
+		EndTime:    end,
```

Comment:
you should add the following to the compaction job record if scheduleID is nonzero:

```
createdBy: CreatedByInfo: &jobs.CreatedByInfo{
				Name: jobs.CreatedByScheduledJobs,
				ID:   int64(sj.ScheduleID()),
			},
```

Reply:
i'd double check there are no side effects for setting this. but this feels very reasonable.

Reply:
Doesn't appear there are any side effects, added!

Reply:
Oof I spoke too soon. One side effect is in [our test driver for scheduled jobs](https://github.com/kev-cao/cockroach/blob/8c4559b5ca20acfa72f9c642b90351df8b340892/pkg/backup/create_scheduled_backup_test.go#L147-L148). Because the compacted job that is created is marked as created by that schedule, the `waitForSuccessfulScheduledJob` ends up waiting on the compaction instead of the incremental backup.

Reply:
Also, because the compaction job is placed under the same schedule, it can prevent backup jobs from running until the compaction job is completed.

Reply:
Perhaps we should add a comment that explains why we intentionally do not set this.